jdbc-adapter/integration-test-data/socker.py

```diff
@@ -30,13 +33,33 @@ def docker_run(config):
                 print(cmd)
                 run(cmd)
 
+def run_with_docker_compose(properties):
+    cmd = "git clone {compose_repo}".format(
+        compose_repo = properties['dockerComposeRepo']
+    )
+    print(cmd)
+    run(cmd)
+    cmd = "echo -e '\nnetworks:\n  default:\n    external:\n      name: {network}\n' >> docker-hive/docker-compose.yml".format(
+        network = DOCKER_NETWORK
+    )
+    print(cmd)
+    run(cmd)
+    cmd = "sudo docker-compose up -d"
+    print(cmd)
+    run(cmd, "./docker-hive")
+    time.sleep(90)
```

Comment:
As you know using `sleep` to wait for a condition is always the second best approach. Any chance to at least assert the condition you are waiting for after the sleep and bail out if it is not fulfilled?

Reply:
You can use `docker-compose ps` or `docker ps` but that the containers are running does not mean that the Hive cluster is ready to accept connections. You would have to observe the log output of the (right) container to know when it is ready. I don know enough about hive to implement this.